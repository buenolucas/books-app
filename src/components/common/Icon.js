import React from 'react';
import PropTypes from 'prop-types';
import {Image} from 'react-native';

const Icon = ({i}) => {
  switch (i) {
    case 'search':
      return <Image source={require('../../assets/icons/search.png')} />;
    case 'filter':
      return <Image source={require('../../assets/icons/filter.png')} />;
    case 'caretLeft':
      return <Image source={require('../../assets/icons/caret-left.png')} />;
    case 'caretRight':
      return <Image source={require('../../assets/icons/caret-right.png')} />;
    case 'close':
      return <Image source={require('../../assets/icons/close.png')} />;
    case 'cancel':
      return <Image source={require('../../assets/icons/cancel.png')} />;
    case 'rightGo':
      return <Image source={require('../../assets/icons/right-go.png')} />;
    case 'rightGoLight':
      return (
        <Image source={require('../../assets/icons/right-go-light.png')} />
      );
  }
};

Icon.propTypes = {
  i: PropTypes.string.isRequired,
};

export default Icon;
