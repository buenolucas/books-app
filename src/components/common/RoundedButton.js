import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import Typography from './Typography';
import Theme from '../../Theme';

const ButtonContainer = styled.TouchableOpacity`
  display: flex;
  flex-direction: row;
  align-items: center;
  width: auto;
  background-color: ${props =>
    props.primary
      ? props.theme.colors.buttonPrimary
      : props.theme.colors.buttonDefault};
  border-radius: ${props => props.theme.spacing.small}px;
  padding: ${({theme}) =>
    theme.spacing.tiny + 'px ' + theme.spacing.small + 'px'};
`;

const RoundedButton = props => {
  const {label, icon, primary, iconPosition, style, onPress} = props;
  const typograpyStyle = {};
  if (primary) {
    typograpyStyle.color = Theme.colors.white;
  }
  if (icon && label) {
    if (iconPosition == 'left') {
      typograpyStyle.marginLeft = Theme.spacing.tiny;
    } else {
      typograpyStyle.marginRight = Theme.spacing.tiny;
    }
  }

  return (
    <ButtonContainer primary={primary} onPress={onPress} style={style}>
      {icon && iconPosition == 'left' && icon}
      {label && (
        <Typography type="caption2" style={typograpyStyle}>
          {label}
        </Typography>
      )}
      {icon && iconPosition == 'right' && icon}
    </ButtonContainer>
  );
};

RoundedButton.propTypes = {
  icon: PropTypes.element,
  label: PropTypes.string,
  primary: PropTypes.bool,
  iconPosition: PropTypes.string,
  style: PropTypes.object,
};

RoundedButton.defaultProps = {
  primary: false,
  iconPosition: 'left',
};

export default RoundedButton;
