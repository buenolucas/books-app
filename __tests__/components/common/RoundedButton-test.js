import React from 'react';
import {shallow} from 'enzyme';
import 'jest-styled-components';
import RoundedButton from '../../../src/components/common/RoundedButton';
import Icon from '../../../src/components/common/Icon';

const pressFn = jest.fn();

describe('RoundedButton', () => {
  describe('Rendering', () => {
    it('renders without crashing', () => {
      const component = shallow(
        <RoundedButton
          label="button render test"
          icon={<Icon i="search" />}
          onPress={pressFn}
        />,
      );
      expect(component).toMatchSnapshot();
    });
  });
});
