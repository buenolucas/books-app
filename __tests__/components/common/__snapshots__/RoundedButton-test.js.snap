// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedButton Rendering renders without crashing 1`] = `
<Styled(TouchableOpacity)
  onPress={[MockFunction]}
  primary={false}
>
  <Icon
    i="search"
  />
  <Typography
    style={
      Object {
        "marginLeft": 8,
      }
    }
    type="caption2"
  >
    button render test
  </Typography>
</Styled(TouchableOpacity)>
`;
